kill.go

```diff
@@ -91,17 +91,31 @@ func kill(containerID, signal string) error {
 		return err
 	}
 
-	// container MUST be running
 	podStatus, err := vc.StatusPod(containerID)
 	if err != nil {
 		return err
 	}
-	if podStatus.State.State != vc.StateRunning {
-		return fmt.Errorf("Container is not running")
-	}
 
 	if len(podStatus.ContainersStatus) != 1 {
-		return fmt.Errorf("ContainerStatus list from PodStatus is wrong, expecting only one ContainerStatus")
+		return fmt.Errorf("ContainerStatus list from PodStatus %s is wrong, expecting only one ContainerStatus", containerID)
+	}
+
+	// container MUST be running
+	if podStatus.ContainersStatus[0].State.State != vc.StateRunning {
+		return fmt.Errorf("Container %s is not running", containerID)
+	}
+
+	// Check status of process
+	running, err := processRunning(podStatus.ContainersStatus[0].PID)
+	if err != nil {
+		return err
+	}
+	if running == false {
+		if err := stopContainer(podStatus); err != nil {
+			return err
+		}
+
+		return fmt.Errorf("Process not running inside container %s", containerID)
```

Comment:
This now looks clearer to me now, but looking at this afresh, the code looks racy: wouldn't it be safer to simply have `kill()` call `vc.StopContainer()` since that will/should handle checks for the pod, container and process?